.github/workflows/release.yml

```diff
@@ -192,40 +71,32 @@ jobs:
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Build & Push Image
-        uses: docker/build-push-action@v5
-        with:
-          context: .
-          file: ./buildscripts/zfs-driver/zfs-driver.Dockerfile
-          push: true
-          platforms: linux/amd64, linux/arm64
-          tags: |
-            ${{ steps.docker_meta.outputs.tags }}
-          build-args: |
-            DBUILD_DATE=${{ steps.date.outputs.DATE }}
-            DBUILD_REPO_URL=https://github.com/openebs/zfs-localpv
-            DBUILD_SITE_URL=https://openebs.io
-            BRANCH=${{ env.BRANCH }}
-
-  release-chart:
-    runs-on: ubuntu-22.04
-    needs: ["csi-driver"]
-    steps:
-      - uses: actions/checkout@v4
+      - name: Mirror multi-arch image with crane to Dockerhub
+        env:
+          SRC: ghcr.io/${{ env.IMAGE_ORG }}/dev/zfs-driver:${{ env.VERSION }}
+          DH: docker.io/${{ env.IMAGE_ORG }}/zfs-driver:${{ env.VERSION }}
+        run: |
+            nix-shell --pure --run "crane copy --platform all ${{ env.SRC }} ${{ env.DH }}" ./shell.nix
 
-      - uses: cachix/install-nix-action@v22
-      - name: Pre-populate nix-shell
+      - name: Mirror multi-arch image with crane to Github
+        env:
+          SRC: ghcr.io/${{ env.IMAGE_ORG }}/dev/zfs-driver:${{ env.VERSION }}
+          GH: ghcr.io/${{ env.IMAGE_ORG }}/zfs-driver:${{ env.VERSION }}
         run: |
-          export NIX_PATH=nixpkgs=$(jq '.nixpkgs.url' nix/sources.json -r)
-          echo "NIX_PATH=$NIX_PATH" >> $GITHUB_ENV
-          nix-shell --pure --run "echo" ./shell.nix
+            nix-shell --pure --run "crane copy --platform all ${{ env.SRC }} ${{ env.GH }}" ./shell.nix
+
+      - name: Mirror multi-arch image with crane to Quay
+        env:
+          SRC: ghcr.io/${{ env.IMAGE_ORG }}/dev/zfs-driver:${{ env.VERSION }}
+          QY: quay.io/${{ env.IMAGE_ORG }}/zfs-driver:${{ env.VERSION }}
+        run: |
+            nix-shell --pure --run "crane copy --platform all ${{ env.SRC }} ${{ env.QY }}"
 
-      - name: Check if the chart is publishable
+      - name: Update the registry and the repository
         run: |
-          TAG=${{ github.event.release.tag_name }}
-          nix-shell --pure --run "./scripts/update-chart-version.sh --tag $TAG --publish-release" ./shell.nix
+          nix-shell --pure --run "./scripts/update-reg-repo.sh --registry docker.io/ --repository ${{ env.IMAGE_ORG }}/zfs-driver" ./shell.nix
```

Comment:
ah we still need to modify the chart! that's unfortunate

Reply:
then we don't need the tar?

Reply:
how about packing the chart on its final form and adding a template function which replaces the registry and namespace if a helm env var is set? ex: `--set zfs.staging=true` ?

Reply:
We can, but any issues doing this way?

Reply:
Either way we need to set the values. Also we would need to expose an extra value?

Reply:
The action doesn't seem to support the bundled tar.